### ISSUE TITLE
Cannot cancel request in a before_request callback

### ISSUE THREAD

**Issue by noctivityinc** [flexirest/flexirest#104]
We're trying to use flexirest with a Auth Bearer authentication.  We're using `ActiveSupport::CurrentAttributes` to store the auth token after an admin is authenticated so that we can access that token in the before_request in our models before making all our API requests.

The problem is that we want to check to make sure that token exists before making each request so that we don't unnecessarily hit the API.  This is normally pretty simple in a callback with a simple `false` but it's not working in this case:

```
before_request do |name, request|
    if Current.token.blank?
      false
    else
      request.headers["Authorization"] = "Bearer: #{Current.token}"
    end
  end
```

Any ideas?

**Comment by andyjeffries** [comment]
This is a good point. I would normally say to raise an exception here though and rescue it, because stopping a callback with "false" doesn't really clarify WHY it failed to (for example) "save the person" to the API.

**Comment by noctivityinc** [comment]
True, but I think it should still support it like other callbacks.

**Comment by andyjeffries** [comment]
This has been implemented now and has been released as v1.7.3. Thanks for the issue.
